### PR TITLE
Use collaborators in assignees list

### PR DESCRIPTION
**Pull request by sbose78**
fixes https://github.com/fabric8io/fabric8-planner/issues/1538

Instead of pulling all users for assignee list, this PR enables the UI to pull only the collaborators. 

The collborators URL is present in the space response. Here's a sample :
https://github.com/almighty/almighty-core/pull/1116#issuecomment-296457081

### CODE REVIEW

**Comment by sbose78** [comment]
Please feel free to close/update this PR if you feel the method needs to go somewhere ( or patiently help me understand where it should go :) :) )

I was unable to test this locally because I couldn't login to planner.

File: src/app/work-item/work-item-detail/work-item-detail.component.ts [hunk 1]
```diff
@@ -324,7 +324,7 @@ export class WorkItemDetailComponent implements OnInit, AfterViewInit, OnDestroy
   getAllUsers(): Observable<any> {
     return Observable.combineLatest(
       this.userService.getUser(),
-      this.userService.getAllUsers()
+      this.workItemService.getCollaborators()
```
Comment (sanbornsen) [inline src/app/work-item/work-item-detail/work-item-detail.component.ts]: We already have a collaborator service. You can use that instead creating a new **getCollaborators** in WI service. This is the file - **src/app/collaborator/collaborator.service.ts**

Comment (sanbornsen) [inline src/app/work-item/work-item-detail/work-item-detail.component.ts]: You can actually update that file to use **.relationships.collaborators.links.related** in line 24

Comment (sbose78) [inline src/app/work-item/work-item-detail/work-item-detail.component.ts]: Thanks a lot, I will first add `collaborators` to https://github.com/fabric8-ui/ngx-fabric8-wit/blob/master/src/app/models/space.ts#L27 and then consume it here ?

Comment (alexeykazakov) [inline src/app/work-item/work-item-detail/work-item-detail.component.ts]: Check with @joshuawilson and @pmuir if ngx module is the right place for that.

Comment (sanbornsen) [inline src/app/work-item/work-item-detail/work-item-detail.component.ts]: @sbose78 yeah that's needed there.

Comment (sbose78) [inline src/app/work-item/work-item-detail/work-item-detail.component.ts]: thanks @sanbornsen , could you please tell me how to fix the tests :)

Comment (sanbornsen) [inline src/app/work-item/work-item-detail/work-item-detail.component.ts]: @sbose78 That's  bit tricky to explain. I'll raise a PR on your fork / this branch.

Comment (sbose78) [inline src/app/work-item/work-item-detail/work-item-detail.component.ts]: @sanbornsen I understand, feel free to clone my branch( this PR ) and make changes here directly. ( No need to open a  PR on this PR/branch )

**Comment by fabric8cd** [comment]
@sbose78 snapshot fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-1734-fabric8-planner.int.rdu2c.fabric8.io

**Comment by sbose78** [comment]
1. Navigated to planner on https://fabric8-ui-fabric8-ui-pr-1734-fabric8-planner.int.rdu2c.fabric8.io/
2. Created a workitem and found that only I was in the list in the "assignee" dropdown. This is because I'm the lone collaborator in my space.
3. Then I went to add a new collaborator using the REST API 
`POST https://api.prod-preview.openshift.io/api/spaces/d38c2a6a-929b-46d6-9b54-19be89515a7d/collaborators/fb82bdf8-e823-446e-9ab0-b78870809ccb`
4. Now when I create a workitem, I see there are 2 users ( me + the one I added as collaborator )

Looks like  things are working as expected.

**Comment by fabric8cd** [comment]
@sbose78 snapshot fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-1734-fabric8-planner.int.rdu2c.fabric8.io

**Comment by sbose78** [comment]
@dgutride is working on the "collaborators" on settings screen : https://github.com/fabric8io/fabric8-ui/pull/1048

After that's in :
the changes in this PR would be easier to use.

**Comment by joshuawilson** [comment]
@sbose78 collaborators PR has been merged in fabric8-ui

**Comment by sbose78** [comment]
Thanks @joshuawilson - merging this would now make sense :)

**Comment by fabric8cd** [comment]
@sbose78 snapshot fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-1734-fabric8-planner.int.rdu2c.fabric8.io

**Comment by sanbornsen** [comment]
@sbose78 shall we merge this?

**Comment by sbose78** [comment]
Yes, please. @sanbornsen .

**Comment by fabric8cd** [comment]
@sbose78 snapshot fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-1734-fabric8-planner.int.rdu2c.fabric8.io

**Pull request merged**